dmutils/content_loader.py

```diff
@@ -413,6 +404,27 @@ def _get_single_question_data(self, form_data):
 
         return {self.id: value if value else None}
 
+    def get_error_messages(self, errors):
+        error_fields = set(errors.keys()) & set(self.form_fields)
+        if not error_fields:
+            return {}
+
+        for field_name in sorted(error_fields):
+            question = self.get_question(field_name)
+            message_key = errors[field_name]
+
+            validation_message = question.get_error_message(message_key, field_name)
+
+            error_key = question.id
+            if message_key == 'assurance_required':
+                error_key = '{}--assurance'.format(error_key)
+
+            return {error_key: {
```

Comment:
I don't think this could ever happen: question.id is acting as error_key for everything except assurance questions, but assurance error is returned in the same field key from the API – so it's impossible to get both value and assurance errors from the API. Multiquestions are acting as sections in form validation, so this doesn't apply to them either.

Having said that, that explanation relies on a lot of current behaviour, so I've changed this to collect all the errors for question fields. This might change the error we end up with for multi-field questions (the last field wins, as opposed to the first one), but that's similar to the previous behaviour anyway.